airflow/executors/kubernetes_executor.py

```diff
@@ -66,7 +67,7 @@ class ResourceVersion:
     """Singleton for tracking resourceVersion from Kubernetes"""
 
     _instance = None
-    resource_version = "0"
+    resource_version: dict[str | None, str | None] = {}
```

Comment:
does this not work?
```suggestion
    resource_version: dict[str, str] = {}
```

Reply:
Hi @dstandish , when `multi_namespace_mode` is `True` & `namespace_list` is not specified, that means the user will still want to apply the way using cluster-scope role. In this case, argument `namespace` will be None for the corresponding `KubernetesJobWatcher`. That's they the key can be `None`.

But the value doesn't have to be `str | None`. So I will change this to `resource_version: dict[str | None, str] = {}`.

Reply:
Changed to `resource_version: dict[str | None, str] = {}` in commit
https://github.com/apache/airflow/pull/28047/commits/a23c33e90bd6666286e03cc676aba9a6322d570c